test/src/TimeoutExtender.spec.js

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) 2015-2017 TechnologyAdvice
+ */
+
+'use strict'
+
+const TimeoutExtender = require('src/TimeoutExtender')
+const SquissStub = require('test/stubs/SquissStub')
+const Message = require('src/Message')
+const delay = require('delay')
+
+let inst = null
+let clock = null
+const fooMsg = new Message({ msg: { MessageId: 'foo', Body: 'foo' } })
+const barMsg = new Message({ msg: { MessageId: 'bar', Body: 'bar' } })
+const bazMsg = new Message({ msg: { MessageId: 'baz', Body: 'baz' } })
+
+describe('TimeoutExtender', () => {
+  afterEach(() => {
+    if (clock && clock.restore) clock.restore()
+    inst = null
+  })
+  it('adds and deletes a message through function calls', () => {
+    inst = new TimeoutExtender(new SquissStub())
+    inst.addMessage(fooMsg)
+    inst._index.should.have.property('foo')
+    inst.deleteMessage(fooMsg)
+    inst._index.should.not.have.property('foo')
+  })
+  it('adds and deletes a message through events', () => {
+    const squiss = new SquissStub()
+    inst = new TimeoutExtender(squiss)
+    const addSpy = sinon.spy(inst, 'addMessage')
+    const delSpy = sinon.spy(inst, 'deleteMessage')
+    squiss.emit('message', fooMsg)
+    return delay(5).then(() => {
+      addSpy.should.be.calledOnce()
+      squiss.emit('handled', fooMsg)
+      return delay(5)
+    }).then(() => {
+      delSpy.should.be.calledOnce()
+    })
+  })
+  it('fails silently when asked to delete a nonexistent message', () => {
+    inst = new TimeoutExtender(new SquissStub())
+    inst.addMessage(fooMsg)
+    inst.deleteMessage(barMsg)
+  })
```

Comment:
Interesting... no assertion, but guessing this would fail if it wasn't "silent" like you are expecting?